### ISSUE TITLE
Steam segfaults for no apparent reason under XUbuntu 2021.04 after some time

### ISSUE THREAD

**Issue by DJGummikuh** [ValveSoftware/steam-for-linux#7820]
#### Your system information

* Steam client version (build number or date): May 19 2021, 01:54:04 / 1621394999
* Distribution (e.g. Ubuntu): XUbuntu
* Opted into Steam client beta?: Yes
* Have you checked for system updates?: Yes
* nVidia Drivers: nvidia-driver-460 proprietary

#### Please describe your issue in as much detail as possible:
Expected: Steam continues running and doesn't crash without warning.
Actual: Steam starts normally, then after *some* time, it just disappears. See https://gist.github.com/DJGummikuh/05dd2a8111fedf5cbe1b1aa0109d98e1 for terminal log output. I already deleted steam and reinstalled it, which caused no difference in behavior. I *presume* that this has to do with my update to Xubuntu 2021.04 but have no conclusive way to prove that.

#### Steps for reproducing this issue:

1. Start Steam
2. Start a game (or not, doesn't seem to make a difference)
3. Wait
4. See that steam crashes, usually tearing down the game alongside.

**Comment by keddad** [comment]
I have the same issue on Arch-based distribution. I've uploaded the [logs](https://gist.github.com/keddad/eef7b322a2ee866f4170667788b6ecaf) of Steam running with GDB attached, as well as the [dump](https://drive.google.com/file/d/1JeJKJ_CHBtg-RvvLe1K54o079bN7DekG/view?usp=sharing) file. I'm running `nvidia-465.31-4` driver, same Steam version.

**Comment by keddad** [comment]
This might be connected to #6751 : I disabled Docker service, and Steam worked flawlessly for at least 40 minutes

**Comment by DJGummikuh** [comment]
Good catch, I actually do have docker running on my system. Stopped it now and waiting if steam crashes again
